docs/samples/graph/README.md

```diff
@@ -0,0 +1,251 @@
+  - [**3 Examples**](#3-examples)
+    - [**3.1 Single**](#31-single)
+    - [**3.2 Switch**](#32-switch)
+    - [**3.3 Splitter**](#33-splitter)
+    - [**3.4 Ensemble**](#34-ensemble)
+# **Inference Graph**
+## **1. Problem** 
+
+Often the time on a production inference pipeline, it is not a single inference service, models need to be chained together to produce the final prediction result. For example a face recognition pipeline may need to find the face area first and then compute the features of the faces to match the face database, these two models are depending on each other and the first model’s output is the second model’s input. Another example is the NLP pipeline, it is very common that you would need to do some document classification first followed by downstream named entity detection or text summary tasks.[[...]](https://docs.google.com/document/d/13VHfOxa72pgoy5Eg5c-gGHZfEL7uBDc7rTLa2pAi4Ko/edit#heading=h.x9snb54sjlu9)
+
+KServe inference graph is designed for this.
+
+## **2. KServe Inference Graph** 
+![image](graph.png)
+### **2.1 Inference Graph**
+As above image shows, inference graph is made up of a list of `nodes` , and each `node` consists of several `isvcs`. Every graph must have a root node named `root`. When an inference request hits the graph it excutes the `root` node from the DFS, if the graph has other `nodes`, it will pass the `$request` or `$response` of the root service as input data to the `next node`. There are four types of `node` are supported: ***Single***, ***Switch***, ***Ensemble***, ***Splitter***.
+
+
+### **2.2 Single Node**
+**Single Node** makes user can connect 2 isvcs in a sequence relationship. The
+`routes` field defines the first `isvc`, and if this node is not the tail node of the graph, it will have one `nextRoute` as the second `isvc`. User can choose `$request` or `$response` from the first `isvc` as the input data to send to the second `isvc`.
+
+![image](singleNode.png)
+```yaml
+...
+root:
+  routerType: Single 
+  routes:
+  - service: isvc1
+  nextRoutes:
+  - nodeName: isvc2
+    data: $request
+...
+```
+### **2.3 Switch Node**
+**Switch Node** makes user can select an `isvc` to handle the request by setting the `condition`. Usually, user doesn't need to set the `serviceURL`, kserve will fill it with `isvc.status.address.URL` of `isvc`, but if you want to specified the `serviceURL`, you can set it manually. 
+
+![image](switchNode.png)
+```yaml
+mymodel:
+  routerType: Switch
+  routes:
+  - service: isvc1
+    serviceUrl: http://isvc1.default.example.com/splitter
+    condition: "{.target == \"blue\"}"
+  - service: isvc2
+    serviceUrl: http://isvc2.default.example.com/splitter
+    condition: "{.target != \"blue\"}"
+  nextRoutes:
+  - nodeName: isvcM
+    data: $response
+```
+We use `k8s.io/client-go/util/jsonpath` to parse the condition, Because of the kubernetes jsonpath can not support `map` fillter, so we make a little changes on the input data and `condition`:
```

Comment:
```suggestion
We use `k8s.io/client-go/util/jsonpath` to parse the condition, because the kubernetes jsonpath can not support `map` input, we make a change on the input data and `condition`:
```